homeassistant/components/overkiz/config_flow.py

```diff
@@ -144,9 +140,15 @@ async def async_step_zeroconf(
         gateway_id = properties["gateway_pin"]
 
         LOGGER.debug("ZeroConf discovery detected gateway %s", obfuscate_id(gateway_id))
+        return await self._process_discovery(gateway_id)
 
+    async def _process_discovery(self, gateway_id: str) -> FlowResult:
+        """Handle discovery of a gateway."""
         await self.async_set_unique_id(gateway_id)
         self._abort_if_unique_id_configured()
+        self.context["title_placeholders"] = {
+            CONF_NAME: f"Gateway: {obfuscate_id(gateway_id)}"
```

Comment:
I don't think we need to obfuscate the id as its being published to the local network.  Adding obfuscation where it isn't needed gives the false impression that its a security control which can lead to some bad assumptions about how the system is supposed to work.

Reply:
We actually talked about it and I added it here as it's obfuscated in the logs. But I agree it's not necessary.

Reply:
@bdraco the reason we obfuscated the ID is because this is also used as the gateway 'pin', however I guess people won't take screenshots of their discovery info and put it in an issue (or another public place).

Having the full id should be fine, and if many people share it, we can always obfuscate it indeed.

Reply:
That seems like a poor design to not make the pin something that isn't being broadcast over the network.  🤷

Reply:
If you prefer to leave it in, please document it with a comment that it is not intended to be an effective security measure.